test/integration/build_admission_test.go

```diff
@@ -53,6 +54,9 @@ func TestPolicyBasedRestrictionOfBuildStrategies(t *testing.T) {
 		t.Errorf("unexpected error: %v", err)
 	}
 
+	// Give the policy cache a chance to catch up
+	time.Sleep(time.Second)
```

Comment:
I'd rather have an "expectedCondition" function (I really want a type, but you'll probably say no) passed to a bounded runUntil like structure.

Actually, what I really really want, is a way a check to see if the policy cache is current against the live etcd, but I don't have any good thoughts on how to expose that.